packages/serverpod/lib/src/server/endpoint_dispatch.dart

```diff
@@ -307,8 +276,20 @@ abstract class EndpointMethodConnector {
   /// List of parameters used by the method.
   final Map<String, ParameterDescription> params;
 
+  /// List of [Scope]s that are required to access this [EndpointMethodConnector].
+  Set<Scope> requiredScopes;
+
+  /// States if the [EndpointMethodConnector] only should accept users that are authenticated.
+  /// Default value is false
+  bool requireLogin;
```

Comment:
Yes, but you better make it final so that is never mutated after the instance has been created.

Reply:
Same for required scopes.

Could even be an idea that this is just straight up delegated to the EndpointConnector directly instead of storing a cached value.

Reply:
Shouldn't this change be reflected in the generator as well? So that the constructor of the EndpointMethodConnector is correctly initiated?

Reply:
In my mind, all our authentication tests for streaming methods should fail with this change?

Reply:
Your comments here make me think I completely misunderstood what you suggested the other day then 😄 Lets discuss!

Reply:
Refactored to return a call context that exposes `endpoint` and thereby these values: [8573727](https://github.com/serverpod/serverpod/pull/2676/commits/8573727c0e0f74a5ceda85e4d518ae9a005b737b)